oolab/src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -1,19 +1,23 @@
 package agh.ics.oop;
 import java.util.HashMap;
 
-abstract class AbstractWorldMap implements IWorldMap,IPositionChangeObserver  {
+public abstract class AbstractWorldMap implements IWorldMap,IPositionChangeObserver  {
 
     protected HashMap<Vector2d, Animal> animalsHashMap = new HashMap<>();
+    protected MapBoundary mapBoundary = new MapBoundary();
+
 
 
     @Override
-    public boolean place(Animal animal) {
+    public boolean place(Animal animal) throws IllegalArgumentException {
```

Comment:
deklaracja `throws` niepotrzebna, bo `IllegalArgumentException` to tzw. unchecked exception.